kong/plugins/aws-lambda/handler.lua

```diff
@@ -23,28 +27,71 @@ function AWSLambdaHandler:new()
   AWSLambdaHandler.super.new(self, "aws-lambda")
 end
 
-local function retrieve_parameters()
+local function retrieve_parameters(content_type, conf)
   ngx_req_read_body()
   local body_parameters, err
-  local content_type = ngx_req_get_headers()[CONTENT_TYPE]
   if content_type and string_find(content_type:lower(), "multipart/form-data", nil, true) then
     body_parameters = Multipart(ngx_req_get_body_data(), content_type):get_all()
-  elseif content_type and string_find(content_type:lower(), "application/json", nil, true) then
+  elseif content_type and string_find(content_type:lower(), "application/json", nil, true)
+         and not conf.request_body then
     body_parameters, err = cjson.decode(ngx_req_get_body_data())
     if err then
       body_parameters = {}
     end
-  else
+  elseif content_type and string_find(content_type:lower(), "application/x-www-form-urlencoded", nil, true) then
     body_parameters = public_utils.get_post_args()
+  else
+    body_parameters = {}
   end
 
   return utils.table_merge(ngx_req_get_uri_args(), body_parameters)
 end
 
+local function retrieve_body(content_type)
+  ngx_req_read_body()
```

Comment:
I think we should move it up, then, so we aren't needlessly calling multiple times :)